traits/trait_types.py

```diff
@@ -1783,8 +1789,7 @@ def _set(self, object, name, value):
                         or (self._exclude_high and (high > new_value))
                         or ((not self._exclude_high) and (high >= new_value))
                     ):
-                        self._set_value(object, name, new_value)
-                        return
+                        return new_value
             except:
                 pass
```

Comment:
If this branch is reached, the function returns `None`, and `_set` will set `None` to the trait. Previously it skip setting the trait. Perhaps `_validate` should raise here and the `_set` silences the error (to preserve existing behaviour)?

Reply:
I'm confused; can you clarify? Do you mean that the current `_validate` method can return `None`? I can't see a path where that can happen.

Reply:
Ah I failed not see the `self.error(object, name, value)` after this block. Sorry about that.